### DISCUSSION TITLE
Openssl speed reports error for RSA KEM keys using external providers in openssl3.2

### DISCUSSION THREAD

**Discussion by abasapax** [openssl/openssl#24311] [Q&A]
Issue Details:
Openssl version: openssl-3.2.1

When run with external provider, Openssl speed reports the below error for KEM keys in RSA algo.

Error log:
WARNING: the error queue contains previous unhandled errors.
40E776F7FF7F0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto/evp/evp_fetch.c:355:Global default library context, Algorithm (RSA : 106), Properties (null)
40E776F7FF7F0000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto/evp/evp_fetch.c:355:Global default library context, Algorithm (RSA : 106), Properties (null)

Analysis:
1. KEM keys are initialized from "evp_kem_init" via EVP_PKEY_encapsulate_init & EVP_PKEY_decapsulate_init
2. The above errors are observed in speed test before the start of Kem keys performance OPs i.e after KEM keys are initialized
3. On further debug into the issue found that **store management returns method as NULL from "inner_evp_generic_fetch"** code, looks like the default methods for KEM provider are not loaded but not in the case of signature Keys.

Note: In the above test, we are not supporting RSA Keymgmt instead expect openssl to generate the keys for operations.

Please resolve the above issue.

**Comment by t8m** [comment]
You need to at least support key import in the keymgmt in your provider. Also how do you select the provider to be used with speed? You will need to load both your and default providers and prefer your provider via a property query.
